js/view.js

```diff
@@ -0,0 +1,172 @@
+        return {
+            x: coords[0],
+            y: coords[1]
+        };
+    }
+
+    function getTileFromPoint(point) {
+        var selector = '.tile[data-point="' + point.x + ':' + point.y + '"]';
+
+        var tile = document.querySelector(selector);
+        if (!tile) {
+            throw new Error('Expected to find a tile for point:', point, 'selector:', selector);
+        }
+        return tile;
+    }
+
+    function tileMouseOver(point, event) {
+        if (isClearing) { FBE.viewmodel.clearPosition(point); }
+        if (isDrawing) { FBE.viewmodel.tryPlaceSelectedItem(point); }
+
+        var preview = document.getElementById("mousePreview");
+        var location = event.currentTarget.getBoundingClientRect();
+        preview.style.left = location.left + 'px';
+        preview.style.top = location.top + 'px';
+
+        if (FBE.viewmodel.couldPlaceSelectedItem(point)) {
+            preview.classList.remove('blocked');
+        } else {
+            preview.classList.add('blocked');
+        }
+    }
+
+    function createSideBar() {
+        var grid = document.getElementById('sidebar');
+        FBE.viewmodel.getPlaceableItems()
+            .map(createSideBarItem)
+            .forEach(grid.appendChild.bind(grid));
+        FBE.viewmodel.onItemSelected(updatePreview);
+    }
+
+    function createSideBarItem(placeable) {
+        var item = document.createElement("div");
+        item.classList.add("item");
+        item.setAttribute("title", placeable.name);
+        item.addEventListener('click',
+            FBE.viewmodel.pickPlacable.bind(null, placeable));
+        var div = document.createElement("div");
+        div.appendChild(createImage(placeable));
+        item.appendChild(div);
+        return item;
+    }
+
+    function createImage(placeable) {
+        var img = document.createElement("img");
+        img.src = placeable.iconUrl;
+        img.classList.add('pixelated-image');
+        img.setAttribute("alt", placeable.name);
+        return img;
+    }
+
+    function updatePreview(placeable) {
+        var previewContainer = document.getElementById("preview");
+        var previewCopies = document.getElementsByClassName('preview__copy');
+
+        var updateContainer = function (parent) {
+            parent.innerHTML = '';
+            if (placeable) {
+                parent.appendChild(createPreview(placeable));
+            }
+        };
+
+        updateContainer(previewContainer);
+        for (var i = 0; i < previewCopies.length; i++) {
+            updateContainer(previewCopies[i]);
+        }
+    }
+
+    function createPreview(placeable) {
+        var div = document.createElement("div");
+        div.classList.add('preview');
+        div.style.width = placeable.width * 32 - 2 + "px";
+        div.style.height = placeable.height * 32 - 2 + "px";
+        if (placeable.canRotate) {
+            var rotation = (placeable.direction - placeable.defaultDirection);

```

Comment:
seems like the brackets are unneeded

Reply:
agreed; at some point I was trying to work it out with `%` but that was a dead end.